cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -532,9 +528,6 @@ impl Node<Option<cst::VariableDef>> {
             let eref = rel_expr.to_ref_or_slot(var)?;
             match (op, &vardef.entity_type) {
                 (cst::RelOp::Eq, None) => Ok(PrincipalOrResourceConstraint::Eq(eref)),
-                (cst::RelOp::Eq, Some(_)) => Err(self.to_ast_err(ToASTErrorKind::InvalidIs(
```

Comment:
This will lose a useful help message for some errors.

Reply:
Here's an example of the change:
```
msg: "`is` cannot appear in the scope at the same time as `==`"
help: "try moving `is` into a `when` condition"
```
```
msg: "invalid policy scope constraint",
help: "policy scope constraints can only include `==`, `in`, `is`, or `_ is _ in _`"
```

My feeling was that the new message was as helpful as the old one, but happy to be overruled. Maybe it would be best to combine the original error message with the new help text?

Reply:
I feel like it's worth having a bit of extra code to specialize the message of `is` since it can sometimes appear in the scope

Reply:
Also, #564 if it's easy to fix while you're in this code messing about with that error variant

Reply:
Nice call out. I'll make the changes in this thread on Monday.

Reply:
Had some time today. Here are examples of error messages in the latest commit:

```
permit(principal > User::"alice", action, resource);
```
**Error message:** invalid operator in the policy scope: >
**Help text:** policy scope clauses can only use `==`, `in`, `is`, or `_ is _ in _`

```
permit(principal, action != Action::"view", resource);
```
**Error message:** invalid operator in the action scope: !=
**Help text:** action scope clauses can only use `==` or `in`

```
permit(principal is User == User::"Alice", action, resource);
```
**Error message:** `is` cannot be used together with `==`
**Help text:** try using `_ is _ in _`

```
permit(principal, action is Action, resource);
```
**Error message:** `is` cannot appear in the action scope
**Help text:** try moving `action is ..` into a `when` condition

Reply:
Looks good!